packages/server/modules/core/tests/favoriteStreams.spec.js

```diff
@@ -0,0 +1,294 @@
+/* instanbul ignore file */
+const expect = require('chai').expect
+
+const { buildApolloServer } = require('@/app')
+const { StreamFavorites, Streams, Users, knex } = require('@/modules/core/dbSchema')
+const { Roles, AllScopes } = require('@/modules/core/helpers/mainConstants')
+const { createStream } = require('@/modules/core/services/streams')
+const { createUser } = require('@/modules/core/services/users')
+const { addLoadersToCtx } = require('@/modules/shared')
+const { truncateTables } = require('@/test/hooks')
+const { gql } = require('apollo-server-express')
+
+/*
+ * TODO: Extra tests
+ * - Test for new error handler, checking if exceptions are reported correctly in responses
+ */
+
+/**
+ * Cleaning up relevant tables
+ */
+async function cleanup() {
+  await truncateTables([StreamFavorites.name, Streams.name, Users.name])
+}
+
+const favoriteMutationGql = gql`
+  mutation ($sid: String!, $favorited: Boolean!) {
+    streamFavorite(streamId: $sid, favorited: $favorited) {
+      id
+      favoritedDate
+      favoritesCount
+    }
+  }
+`
+
+const favoriteStreamsQueryGql = gql`
+  query ($cursor: String, $limit: Int! = 10) {
+    user {
+      id
+      favoriteStreams(cursor: $cursor, limit: $limit) {
+        totalCount
+        cursor
+        items {
+          id
+        }
+      }
+    }
+  }
+`
+
+describe('Favorite streams', () => {
+  const myPubStream = {
+    name: 'My Stream 1',
+    isPublic: false
+  }
+  const myStream = {
+    name: 'My Stream 2',
+    isPublic: true
+  }
+  const notMyStream = {
+    name: 'Not My Stream 1',
+    isPublic: false
+  }
+  const notMyPubStream = {
+    name: 'Not My Stream 2',
+    isPublic: true
+  }
+  let me = {
+    name: 'Itsa Me',
+    email: 'me@gmail.com',
+    password: 'sn3aky-1337-b1m'
+  }
+  let otherGuy = {
+    name: 'Some Other DUde',
+    email: 'otherguy@gmail.com',
+    password: 'sn3aky-1337-b1m'
+  }
+
+  before(async function () {
+    await cleanup()
+
+    // Seeding
+    await Promise.all([
+      createUser(me).then((id) => (me.id = id)),
+      createUser(otherGuy).then((id) => (otherGuy.id = id))
+    ])
+
+    await Promise.all([
+      createStream({ ...myPubStream, ownerId: me.id }).then((id) => (myPubStream.id = id)),
+      createStream({ ...myStream, ownerId: me.id }).then((id) => (myStream.id = id)),
+      createStream({ ...notMyStream, ownerId: otherGuy.id }).then((id) => (notMyStream.id = id)),
+      createStream({ ...notMyPubStream, ownerId: otherGuy.id }).then(
+        (id) => (notMyPubStream.id = id)
+      )
+    ])
+  })
+
+  after(async () => {
+    await cleanup()
+  })
+
+  describe('when authenticated', () => {
+    /** @type {import('apollo-server-express').ApolloServer} */
+    let apollo
+
+    const favoriteStream = async (sid, favorited) =>
+      await apollo.executeOperation({
```

Comment:
This is the way Apollo suggests writing integration tests
https://www.apollographql.com/docs/apollo-server/testing/testing/